libffi-rs/src/high/mod.rs

```diff
@@ -78,6 +76,17 @@ pub use types::{CType, Type};
 pub mod call;
 pub use call::*;
 
+macro_rules! abort_on_panic {
+    ($msg:literal, $body:expr) => {
+        std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| { $body }))
+            .unwrap_or_else(|err| {
+                std::mem::forget(err); // let's not run any panic payload drop glue
```

Comment:
@RalfJung What's the motivation for this? The comment doesn't really make it clear why this is necessary instead of using e.g. `.unwrap_or_else(|_| ...)`.

Reply:
Honestly I don't know if it is needed. I just know that recently there were some critical issues in the standard library related to panicking payloads returned from `catch_unwind` and I don't know what exactly it is one has to do to defend against that but since we want to abort anyway this seemed like the safest option...

Reply:
Specifically, *if* `_` drops the argument at the top of the function (I don't know if it does), then that drop might panic, and then that would skip the `abort` and continue unwinding.

Using `|_err| ...` would work since then it will definitely get dropped at the end of the function, but using explicit `forget` seemed safer than that.

Reply:
@RalfJung In that case it's best to add a comment above the `forget()` line explaining this more clearly (i.e. basically what you said in https://github.com/tov/libffi-rs/pull/58#discussion_r960045562). The current comment explains _what_ we're doing, but not _why_ :smiley: